commitrmnocb/types.go

```diff
@@ -0,0 +1,270 @@
+		if err != nil {
+			// TODO: lggr
+			continue
+		}
+
+		// MerkleRoots
+		for _, merkleRoot := range obs.MerkleRoots {
+			aggObs.MerkleRoots[merkleRoot.ChainSel] =
+				append(aggObs.MerkleRoots[merkleRoot.ChainSel], merkleRoot)
+		}
+
+		// GasPrices
+		for _, gasPriceChain := range obs.GasPrices {
+			aggObs.GasPrices[gasPriceChain.ChainSel] =
+				append(aggObs.GasPrices[gasPriceChain.ChainSel], gasPriceChain.GasPrice)
+		}
+
+		// TokenPrices
+		for _, tokenPrice := range obs.TokenPrices {
+			aggObs.TokenPrices[tokenPrice.TokenID] =
+				append(aggObs.TokenPrices[tokenPrice.TokenID], tokenPrice.Price)
+		}
+
+		// OnRampMaxSeqNums
+		for _, seqNumChain := range obs.OnRampMaxSeqNums {
+			aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OnRampMaxSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// OffRampNextSeqNums
+		for _, seqNumChain := range obs.OffRampNextSeqNums {
+			aggObs.OffRampNextSeqNums[seqNumChain.ChainSel] =
+				append(aggObs.OffRampNextSeqNums[seqNumChain.ChainSel], seqNumChain.SeqNum)
+		}
+
+		// FChain
+		for chainSel, f := range obs.FChain {
+			aggObs.FChain[chainSel] = append(aggObs.FChain[chainSel], f)
+		}
+	}
+
+	return aggObs
+}
+
+// ConsensusObservation holds the consensus values for all chains across all observations in a round
+type ConsensusObservation struct {
+	// A map from chain selectors to each chain's consensus merkle root
+	MerkleRoots map[cciptypes.ChainSelector]cciptypes.MerkleRootChain
+
+	// A map from chain selectors to each chain's consensus gas prices
+	GasPrices map[cciptypes.ChainSelector]cciptypes.BigInt
+
+	// A map from token IDs to each token's consensus price
+	TokenPrices map[types.Account]cciptypes.BigInt
+
+	// A map from chain selectors to each chain's consensus OnRamp max sequence number
+	OnRampMaxSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus OffRamp next sequence number
+	OffRampNextSeqNums map[cciptypes.ChainSelector]cciptypes.SeqNum
+
+	// A map from chain selectors to each chain's consensus f (failure tolerance)
+	FChain map[cciptypes.ChainSelector]int
+}
+
+// GasPricesArray returns a list of gas prices
+func (co ConsensusObservation) GasPricesArray() []cciptypes.GasPriceChain {
+	gasPrices := make([]cciptypes.GasPriceChain, 0, len(co.GasPrices))
+	for chain, gasPrice := range co.GasPrices {
+		gasPrices = append(gasPrices, cciptypes.NewGasPriceChain(gasPrice.Int, chain))
+	}
+
+	return gasPrices
+}
+
+// TokenPricesArray returns a list of token prices
+func (co ConsensusObservation) TokenPricesArray() []cciptypes.TokenPrice {
+	tokenPrices := make([]cciptypes.TokenPrice, 0, len(co.TokenPrices))
+	for tokenID, tokenPrice := range co.TokenPrices {
+		tokenPrices = append(tokenPrices, cciptypes.NewTokenPrice(tokenID, tokenPrice.Int))
+	}
+
+	return tokenPrices
+}
+
+type OutcomeType int
+
+const (
+	ReportIntervalsSelected OutcomeType = iota + 1
+	ReportGenerated
```

Comment:
nit: the report doesn't generate until after the outcome. I'd also consider removing the `Report` prefix from all of these.
```suggestion
	ReportGeneration
```